packages/amplify_datastore_plugin_interface/lib/src/types/models/subscription_event.dart

```diff
@@ -24,20 +24,19 @@ class SubscriptionEvent<T extends Model> {
   final ModelType<T> modelType;
   final EventType eventType;
 
-  const SubscriptionEvent({this.item, this.modelType, this.eventType});
+  const SubscriptionEvent(
+      {required this.item, required this.modelType, required this.eventType});

```

Comment:
Is this also a pre-existing bug? If not, why are we switching these three params from optional to required?

Reply:
Based on how this class was used it appeared that the fields were necessary. 

Looking at the usage of that constructor you can see only the fromMap constructor calls it and passes all fields. 

`  const SubscriptionEvent(
      {required this.item, required this.modelType, required this.eventType});

  factory SubscriptionEvent.fromMap(
      Map<dynamic, dynamic> map, ModelType<T> modelType) {
    Map<String, dynamic> serializedItem =
        Map<String, dynamic>.from(map["item"]);

    return SubscriptionEvent(
        item: modelType.fromJson(
            Map<String, dynamic>.from(serializedItem["serializedData"])),
        eventType: EventType.values
            .firstWhere((e) => describeEnum(e) == map["eventType"] as String?),
        modelType: modelType);
  }
}
`